### ISSUE TITLE
Use salloc if you want to srun while on a node

### ISSUE THREAD

**Issue by suranap** [StanfordLegion/sapling-guide#1]
I used `srun` to hop into a bash shell on a GPU machine. Then I wanted to use `srun`.to launch 4 processes on this same machine. It just hangs. Looks like `srun` reserves the whole node, and then further calls to `srun` are stuck. So this is a use case for `salloc`, and that's how I do stuff on Frontier/Perlmutter. However, on those systems salloc will jump into the machine also. That's convenient. 

https://github.com/StanfordLegion/sapling-guide/blob/4a2dc09b4fc871a06a14425e3118b476aefc8024/README.md?plain=1#L63-L78

**Comment by elliottslaughter** [comment]
You're right that you'd need to `salloc` and then `srun` inside of that if you want to do multiple jobs inside of an allocation.

Is there a specific request you're making or improvement you suggest? `srun` is the shortest one-line command, so it's what I generally recommend, and doing multiple jobs is generally a special case.

**Comment by suranap** [comment]
Sapling could match the behavior of `salloc` on HPCs by adding [this](https://slurm.schedmd.com/slurm.conf.html#OPT_use_interactive_step) to /etc/slurm.conf:
```
LaunchParameters=use_interactive_step
```
See [here](https://slurm.schedmd.com/faq.html#prompt) for more info. This is now the recommended way to do things.
